device/transport/mqtt/src/mqtt.ts

```diff
@@ -277,6 +277,12 @@ export class Mqtt extends EventEmitter implements DeviceTransport {
               topic += '/';
             }
 
+            // This will not catch all messages that exceed IoT Hub limits because properties contribute the size as well.
```

Comment:
>properties contribute the size as well. [](http://example.com/codeflow?start=83&length=39)

You can account for properties by adding len(topic). Then you're checking everything except the MQTT packet overhead (which might be a constant size)

Reply:
Nope, it's more complicated than that. From my testing, everything up to and including `/events/` in the topic string doesn't count. In the key-value pairs in the topic string for properties, the `=` characters don't count. I think that's enough to know that the service calculation isn't so simple.